DQMOffline/Trigger/plugins/PhotonMonitor.cc

```diff
@@ -54,8 +54,8 @@ PhotonMonitor::PhotonMonitor( const edm::ParameterSet& iConfig ) :
 
 PhotonMonitor::~PhotonMonitor()
 {
-  if (num_genTriggerEventFlag_) delete num_genTriggerEventFlag_;
-  if (den_genTriggerEventFlag_) delete den_genTriggerEventFlag_;
+  if (num_genTriggerEventFlag_) num_genTriggerEventFlag_.release();
```

Comment:
This leaks memory. `release()` returns the pointer and stops the ownership`. Instead, just remove lines 57 and 58. In fact, you can do away with the entire destructor since the compiler will do the right thing.